scalac-scoverage-runtime/shared/src/main/scala/scoverage/Invoker.scala

```diff
@@ -59,6 +66,78 @@ object Invoker {
     }
   }
 
+  /**
+   * We record that the given id has been invoked by appending its id to the coverage
+   * data file.
+   *
+   * This will happen concurrently on as many threads as the application is using,
+   * so we use one file per thread, named for the thread id.
+   *
+   * This method is not thread-safe if the threads are in different JVMs, because
+   * the thread IDs may collide.
+   * You may not use `scoverage` on multiple processes in parallel without risking
+   * corruption of the measurement file.
+   *
+   * Same method as above but added additional features to make it work with
+   * https://github.com/scoverage/scalac-scoverage-plugin/issues/265. Typically, we clean the dataDir
+   * and add coverage instruments to the measurement file at runtime rather than compile time. This is
+   * needed as the runtime process may be running on a different host than the compile time one.
+   * However, since the instruments are on the classpath, we can retrieve the instruments from different hosts.
+   *
+   * @param id the id of the statement that was invoked
+   * @param instrumentsDir the directory where the instrument data is held and needs to be copied from
+   */
+  def invokedWriteToClasspath(id: Int, instrumentsDir: String): Unit = {
+    // Get the output data directory from the environment variable or write to a subdir.
```

Comment:
Computing the dataDir like this is likely to be expensive. You should probably move this out into a `@volatile var`, and compute it once at startup.

That would also allow/require you to mutate it in unit tests.